crates/flux-fhir-analysis/src/lib.rs

```diff
@@ -207,10 +227,30 @@ fn assoc_refinement_def(
 
     if let Some(assoc_reft) = assoc_reft {
         let wfckresults = genv.check_wf(impl_id)?;
-        Ok(rty::EarlyBinder(conv::conv_assoc_reft_def(genv, assoc_reft, &wfckresults)?))
+        Ok(Some(rty::EarlyBinder(conv::conv_assoc_reft_def(genv, assoc_reft, &wfckresults)?)))
     } else {
-        Err(QueryErr::InvalidAssocReft { impl_id: impl_id.to_def_id(), name })?
+        Ok(None)
+    }
+}
+
+fn assoc_refinement_def(
+    genv: GlobalEnv,
+    impl_id: LocalDefId,
+    name: Symbol,
+) -> QueryResult<rty::EarlyBinder<rty::Lambda>> {
+    let trait_id = genv.tcx().trait_id_of_impl(impl_id.to_def_id());
```

Comment:
re https://github.com/flux-rs/flux/pull/781/files#r1753073697: We should use the _resolved id_ here, i.e., if `impl_id` is for an extern spec we should use the def_id of the extern item. Using `MaybeExternId` makes it a bit easier to catch these errors. There's no hard rule, but in general, one should use the resolved id when calling queries either in `tcx` or `genv`.